bluesky/preprocessors.py

```diff
@@ -289,6 +290,24 @@ def print_summary_wrapper(plan):
         cmd = msg.command
         if cmd == 'open_run':
             print('{:=^80}'.format(' Open Run '))
+            # This next bit checks the metadata for json compatibility
+            try:
+                json.dumps(msg.kwargs)
+            except TypeError:
+                error_keys=[]
```

Comment:
If we are going to go down the route of sorting out what is failing, might as well write this as a recursive helper function.  The start document can be arbitrarily nested so the error checking should be too.

Reply:
I will make this change, it makes sense. I just have to think about how to keep track fo the nesting when reporting back.